File: adapters/elixir/testdata/lib/greeter.ex
defmodule Testproject.Greeter do
  @moduledoc "Behaviour for greeting."

  @callback greet(name :: String.t()) :: String.t()
end

defmodule Testproject.ConsoleGreeter do
  @behaviour Testproject.Greeter

  def greet(name) do
    "Hello, #{name}!"
  end
end

defprotocol Testproject.Printer do
  @doc "Protocol for printing."
  def print(this, value)
end

defimpl Testproject.Printer, for: BitString do
  def print(_this, value) do
    IO.puts(value)
  end
end

File: adapters/go/testdata/pkg/greeter.go
package pkg

// Greeter defines an interface for greeting.
type Greeter interface {
	Greet() string
}

// ConsoleGreeter implements Greeter.
type ConsoleGreeter struct{}

// Greet returns a greeting string via ConsoleGreeter.
func (g *ConsoleGreeter) Greet() string {
	return "hello"
}

// SayHello calls g.Greet() via the Greeter interface.
func SayHello(g Greeter) string {
	return g.Greet()
}
